### PR TITLE
feat: Implement labels

### LINKED ISSUE #13: Labels break highlighting
One of our uses of Groovy is the execution of Spock tests. Spock is a test framework, which makes [extensive use of labels](https://spockframework.org/spock/docs/1.0/spock_primer.html#_blocks) to mark the sections of a test. Example:

```groovy
class Test {
	def "test add"() {
		when:
		def a = 10
		def b = 10
		def result = a + b

		then:
		result == 20
	}
}
```

Labels are kind of "inherited" from [Java](https://docs.oracle.com/javase/specs/jls/se7/html/jls-14.html#jls-14.7). With the current parser, they seem to be missing/broken.
Highlighting for the example in my neovim (Note the broken `def` highlighting in the line right after the first label):

Bildschirmfoto 2024-07-17 um 14 47 30

In bigger files, the highlighting is all over the place.

I tried to locate label parsing in your grammar.js and could not find it, but admittedly, I am not a treesitter expert. Can you verify that labels are simply missing? If they are missing, I might give their implementation a try.

### PR DESCRIPTION
**Pull request by HendrikJanssen**
fixes #13

I did add the `optional(label)` to the statement sequence, also added expression statements since these were missing?

I am *not* really sure what happened with the `for` tests. From a glimpse, these were wrong before? A single `increment_op` was what I was expecting, not sure where the `assignment` came from beforehand, maybe someone can enlighten me what happened here?

I did not add labels to the break right now since I wanted to keep the changes minimal to be better to review. I will add these and [quoted identifiers](https://groovy-lang.org/syntax.html#_quoted_identifiers) once I am more familiar with tree-sitter and have gathered feedback from this PR.

### CODE REVIEW

**Comment by murtaza64** [comment]
The way I designed the grammar I didn't want a plain `_expression` to count as a `_statement`. Not sure off the top of my head if this breaks anything... but if the tests aren't broken, I think it should be fine?

Adding `_expression` as a `_statement` explains why the `for` tests changed--an `assignment` node is inserted to promote `i++` from an expression (`increment_op`) to a statement

**Comment by murtaza64** [comment]
Feel free to try implementing the rest of the label feature, as long as the tests stay passing and highlighting looks fine on typical code files I'm happy to merge it

File: corpus/label.test [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+(source_file
+  (class_definition
+    (identifier)
+    (closure
+      (function_definition
+        (identifier)
+  (parameter_list)
+  (closure
+    (label
+      (identifier))
```
Comment (murtaza64) [inline corpus/label.test:35]: I think the indentation is off here?

**Comment by murtaza64** [comment]
If you fix the indentation I can merge this as-is

**Comment by HendrikJanssen** [comment]
Im going to come back to this PR on the weekend, I have much to do this week. But thank you very much for your time and explanation!

**Comment by murtaza64** [comment]
I managed to fix the indentation with the python script I wrote to help with the tests (`python treesexpy/ts_test_helper.py corpus/label.test` which diffs the output from treesitter with the test and shows you a line by line diff, optionally updating the test)

**Comment by murtaza64** [comment]
Thanks for the PR!

**Pull request merged**